core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1894,24 +1894,8 @@ private[spark] class DAGScheduler(
                   job.numFinished += 1
                   // If the whole job has finished, remove it
                   if (job.numFinished == job.numPartitions) {
-                    markStageAsFinished(resultStage)
```

Comment:
Why `markStageAsFinished` no need?

Reply:
`cancelRunningIndependentStages` already does that, see:

https://github.com/apache/spark/blob/14d854beeabcf872175edfd65cb6475ecb7d0ae7/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L2862-L2863

Reply:
Got it.

Reply:
@mridulm Oh I see. Thanks for catching this.